### PR TITLE
add `force` to `to_cnf` and `to_dnf`; `is_literal` made more general

### LINKED ISSUE #9949: boolalg.py:is_cnf fails when it should pass
- If i create a simple conjunction of two literals, which by definition is in conjunctive normal form, the boolalg.py:is_cnf function incorrectly returns false.
- Additionally, if I create I an expression in disjunctive normal form, and then call boolalg.py:to_cnf on it. It is correctly put in conjunctive normal form, but boolalg.py:is_cnf still returns false.

Below is sample code that demonstrates both of these issues. 

``` Python
from sympy import S
from sympy.logic import boolalg
from sympy import srepr

literal1 = S("X>2")
literal2 = S("X<4")
literal3 = S("Y>-5")

# Conjunction by itself, which should be in CNF form
conjunction_clause = boolalg.And(literal1, literal2)

# Disjunction of a conjunction and literal, should be in DNF
dnf_exp = boolalg.Or(literal3, conjunction_clause)

print(srepr(conjunction_clause))
print(srepr(dnf_exp))

# Expression already in CNF, but is_cnf fails
simple_is_cnf_test = boolalg.is_cnf(conjunction_clause)
print("simple_is_cnf_test", simple_is_cnf_test)

# Expression in DNF
# Expression correctly put in CNF
print("DNF Expression, turned to CNF: \n\t", boolalg.to_cnf(dnf_exp))
dnf_to_cnf_test = boolalg.is_cnf(boolalg.to_cnf(dnf_exp))

assert((simple_is_cnf_test or dnf_to_cnf_test),
                "2 expression in CNF, but is_cnf returns false.")
```

### LINKED ISSUE #18904: `is_cnf(to_cnf(expr))` should be True
As reported [here](https://stackoverflow.com/questions/60740347/sympy-logic-module-assertis-cnfto-cnfexpr-simplify-true-fails) there are expressions which are in cnf form which, when passed to `to_cnf` with `simplify=True` are returned in non-cnf form. This is a diff that addresses the issue:

```diff
diff --git a/sympy/logic/boolalg.py b/sympy/logic/boolalg.py
index dd734ce..d544ea7 100644
--- a/sympy/logic/boolalg.py
+++ b/sympy/logic/boolalg.py
@@ -1714,7 +1714,9 @@ def to_cnf(expr, simplify=False):
         return expr

     if simplify:
-        return simplify_logic(expr, 'cnf', True)
+        new = simplify_logic(expr, 'cnf', True)
+        if is_cnf(new):
+            return new

     # Don't convert unless we have to
     if is_cnf(expr):
```

cf #9949

### PR DESCRIPTION
**Pull request by smichr**
#### References to other Issues or PRs

fixes #9949
fixes #18904

#### Brief description of what is fixed or changed

When `to_cnf` is given an expression with more than 8 variables and `simplify=True` it will not simplify the expression -- and may not return it in cnf form and does so silently. Now, a ValueError is raised if simplification is requested and `force` is not True. The user has the option to not use `simplify` or to use `force`.

In addition, `is_literal` and `is_a_literal` were combined and made to recognize that things like `x < 2` can be treated like a literal. This fixes issue 9949.

Finally, a quick-exit is given to `simplify_logic` to deal with trivial cases that don't need simplification and are already in the correct, most implified form.

#### Release Notes

* logic
    * boolalg
        * `to_cnf`/`to_dnf` (when `simplify=True) require `force=True` if there are more than 8 variables
        * `simplify_logic` recognizes trivial simplified cases
        * `is_literal` can treat `Not` as literal or not by using the `literal_Not` flag

### CODE REVIEW

**Comment by asmeurer** [comment]
What is the motivation for the literal_Not flag? The inclusion of Not is based on a standard definition of "literal".

**Comment by smichr** [comment]
> What is the motivation for the literal_Not flag

Hmm...I must have had a different problem when trying to detect the quick exit case for `simplify_logic`. When I retested now, everything passed. Let's see if the full suite raises any objections...

File: sympy/logic/boolalg.py [hunk 1]
```diff
@@ -2782,6 +2793,16 @@ def simplify_logic(expr, form=None, deep=True, force=False):
     if form not in (None, 'cnf', 'dnf'):
         raise ValueError("form can be cnf or dnf only")
     expr = sympify(expr)
+    # check for quick exit: right form and all args are
```
Comment (smichr) [inline sympy/logic/boolalg.py]: reviewer: please check to see that this is the right logic for quick exit: if an expression is in the right form and all args are literal then there is nothing left to do.

File: sympy/logic/boolalg.py [hunk 2]
```diff
@@ -1977,9 +1983,13 @@ def is_literal(expr):
 
     """
     if isinstance(expr, Not):
-        return not isinstance(expr.args[0], BooleanFunction)
-    else:
-        return not isinstance(expr, BooleanFunction)
+        return expr.args[0].is_Atom
+    elif expr in (True, False) or expr.is_Atom:
+        return True
+    elif not isinstance(expr, BooleanFunction) and all(
```
Comment (smichr) [inline sympy/logic/boolalg.py]: e.g. `x < 3` is treated like a literal

File: sympy/logic/boolalg.py [hunk 3]
```diff
@@ -1977,9 +1983,13 @@ def is_literal(expr):
 
     """
     if isinstance(expr, Not):
-        return not isinstance(expr.args[0], BooleanFunction)
-    else:
-        return not isinstance(expr, BooleanFunction)
+        return expr.args[0].is_Atom
```
Comment (smichr) [inline sympy/logic/boolalg.py]:
```suggestion
        return is_literal(expr.args[0])
```

File: sympy/logic/boolalg.py [hunk 4]
```diff
@@ -1977,9 +1983,13 @@ def is_literal(expr):
...
+    elif expr in (True, False) or expr.is_Atom:
+        return True
+    elif not isinstance(expr, BooleanFunction) and all(
+            map(is_literal, expr.args)):
```
Comment (smichr) [inline sympy/logic/boolalg.py]:
```suggestion
            a.is_Atom for a in expr.args):
```

**Pull request merged**